examples/fillform.js

```diff
@@ -15,12 +15,13 @@ export default function() {
     // Enter login credentials and login
     page.$('input[name="login"]').type('admin');
     page.$('input[name="password"]').type('123');
-    return page.$('input[type="submit"]').click();
-  }).then(() => {
+  }).then(() =>
+    page.$('input[type="submit"]').click()
+  ).then(() =>
     // We expect the above form submission to trigger a navigation, so wait for it
     // and the page to be loaded.
-    page.waitForNavigation();
-
+    page.waitForNavigation()
+  ).then(() => {
```

Comment:
I think we can use `Promise.all()` now to minimize the chances of a `click()`/`waitForNavigation()` race condition, [as per the Playwright docs](https://playwright.dev/docs/api/class-page#page-wait-for-navigation). :tada:

This seemed to work for me:

```javascript
// Goto front page, find login link and click it
page.goto('https://test.k6.io/', { waitUntil: 'networkidle' });
const elem = page.$('a[href="/my_messages.php"]');
elem.click().then(() => {
  // Enter login credentials and login
  page.$('input[name="login"]').type('admin');
  page.$('input[name="password"]').type('123');
  // We expect the form submission to trigger a navigation, so to prevent a
  // race condition, setup a waiter concurrently while waiting for the click
  // to resolve.
  return Promise.all([
    page.waitForNavigation(),
    page.$('input[type="submit"]').click(),
  ]);
}).then(() => {
  check(page, {
    'header': page.$('h2').textContent() == 'Welcome, admin!',
  });
}).finally(() => {
  page.close();
  browser.close();
});
```

Reply:
I've amended this example, other examples that used `click` and `waitForNavigation` and the readme too.

Resolved in: 886eda55893b293ac9ae973e6ee6690cff9adb7d